### DISCUSSION TITLE
get the user(id) who replied to the tweet (id),

### DISCUSSION THREAD

**Discussion by carlaLanyon** [cjbarrie/academictwitteR#298] [Q&A]
I'm trying to create a network between users who tweet in a specific place and those who reply. To do so, I'm using academic R. I have a data frame with tweets (id) and id author and more information, from the specific place between 2010-2022. The data also comes with the information related to reply_count, like_count, retweet_count, and the user replied_ to, among other information. 

I'm trying to get the user(id) who replied to the tweet (id), something like the function get_retweeted_by, but I haven't been possible.

Any suggestions will be great

Best!

**Comment by chainsawriot** [comment]
@carlaLanyon One way to do this is to use the conversation id

``` r
require(academictwitteR)
#> Loading required package: academictwitteR
x <- get_all_tweets(
start_tweets = "2020-12-01T00:00:00Z",
end_tweets = "2021-11-30T00:00:00Z",
query = "hello",
lang = "en",
country = "GB",
is_retweet = NULL,
n = 100,
)
#> Warning: Recommended to specify a data path in order to mitigate data loss when
#> ingesting large amounts of data.
#> Warning: Tweets will not be stored as JSONs or as a .rds file and will only be
#> available in local memory if assigned to an object.
#> query:  hello place_country:GB lang:en 
#> Total pages queried: 1 (tweets captured this page: 497).
#> Total tweets captured now reach 100 : finishing collection.
x[2,]
#>   lang     geo.place_id geo.coordinates.type geo.coordinates.coordinates
#> 2   en 08104612ff4c5274                 <NA>                        NULL
#>       conversation_id entities.urls entities.annotations
#> 2 1465324057277173772          NULL                 NULL
#>                                                      entities.mentions
#> 2 0, 17, 16, 32, Gaius_Didelphus, RichardDawkins, 2830155391, 15143478
#>   entities.hashtags public_metrics.retweet_count public_metrics.reply_count
#> 2              NULL                            0                          0
#>   public_metrics.like_count public_metrics.quote_count
#> 2                         1                          0
#>                                                                                         text
#> 2 @Gaius_Didelphus @RichardDawkins You're flipping erasing our sex-based language. Hello....
#>   author_id               created_at possibly_sensitive                  id
#> 2 257836462 2021-11-29T23:17:23.000Z              FALSE 1465459907717701634
#>                source in_reply_to_user_id               referenced_tweets
#> 2 Twitter for Android           257836462 replied_to, 1465436827947515914
#>   media_keys
#> 2       NULL
y <- get_all_tweets(build_query(conversation_id = x$conversation_id[2]), start_tweets = "2020-12-01T00:00:00Z", end_tweets = "2021-11-30T00:00:00Z")
#> Warning: Recommended to specify a data path in order to mitigate data loss when
#> ingesting large amounts of data.

#> Warning: Tweets will not be stored as JSONs or as a .rds file and will only be
#> available in local memory if assigned to an object.
#> query:   conversation_id:1465324057277173772 
#> Total pages queried: 1 (tweets captured this page: 481).
#> Total tweets captured now reach 100 : finishing collection.
y$author_id
#>   [1] "38949249"            "1071791596650328075" "414343062"          
#>   [4] "1146587586959249408" "1071791596650328075" "40913079"           
#>   [7] "1207685846918455304" "1450544905651118101" "853955353561661441" 
#>  [10] "1131060755673931776" "414343062"           "3439167791"         
#>  [13] "410263056"           "939665575294279682"  "996670904523423744" 
#>  [16] "127102185"           "1240712894217682946" "1206075203731673088"
#>  [19] "4727378794"          "909200196323684354"  "825838629171916800" 
#>  [22] "220493482"           "1440179071"          "1662523627"         
#>  [25] "10294922"            "19886180"            "248479058"          
#>  [28] "1268537700144959488" "1450527237518802949" "414343062"          
#>  [31] "1450544905651118101" "1406608125281587205" "1400439841192656897"
#>  [34] "909200196323684354"  "241736683"           "1349763403657736192"
#>  [37] "1400622608509542400" "1150916268842614784" "814286897946230784" 
#>  [40] "220493482"           "1071791596650328075" "1196542279839879168"
#>  [43] "1450544905651118101" "1150916268842614784" "1400622608509542400"
#>  [46] "1207685846918455304" "1328945091256475648" "1071791596650328075"
#>  [49] "1029717867510546433" "1327597994808659969" "814286897946230784" 
#>  [52] "40913079"            "99739484"            "1196542279839879168"
#>  [55] "1349763403657736192" "909200196323684354"  "10294922"           
#>  [58] "1485699492"          "1274801974869782529" "1071791596650328075"
#>  [61] "939665575294279682"  "1028108300364201984" "19886180"           
#>  [64] "1453488073833172996" "1450544905651118101" "99739484"           
#>  [67] "437607318"           "1207685846918455304" "1296715498755768320"
#>  [70] "1071791596650328075" "1400439841192656897" "1269394780016062464"
#>  [73] "10294922"            "1090511939871039488" "414343062"          
#>  [76] "1450527237518802949" "14419290"            "14296489"           
#>  [79] "1146103597530148864" "1192337791352311808" "1150916268842614784"
#>  [82] "1275846500900311041" "1328945091256475648" "414343062"          
#>  [85] "3461571434"          "1251310905478045697" "1219736197637066753"
#>  [88] "437607318"           "1945270050"          "1480428566"         
#>  [91] "1298764807252791302" "99739484"            "14419290"           
#>  [94] "2997464485"          "1234826498458124288" "248479058"          
#>  [97] "1485699492"          "1071791596650328075" "1333568713828806656"
#> [100] "814286897946230784"
```

<sup>Created on 2022-03-01 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Reply by carlaLanyon** [reply]
Thank you so much @chainsawriot it works perfectly!!! 

The only issue was when I tried to do it in the entire col, didn´t work. Probably because the query is too long (if I consider all the conversation Id from my date frame).
My code:
y <- get_all_tweets(build_query(conversation_id = x$conversation_id[]), start_tweets = "2020-12-01T00:00:00Z", end_tweets = "2021-11-30T00:00:00Z")

Error in make_query(url = endpoint_url, params = params, bearer_token = bearer_token,  : 
  something went wrong. Status code: 400

And when I tried a longer period
y <- get_all_tweets(build_query(conversation_id = x$conversation_id[]), start_tweets = "2010-01-01T00:00:00Z", end_tweets = "2021-01-01T00:00:00Z",n=1000)

Error in make_query(url = endpoint_url, params = params, bearer_token = bearer_token,  : 
  something went wrong. Status code: 431

My bearer is working good
 Any suggestions? or I should do I manually per each conversation id or maybe per groups?

Best!

**Reply by chainsawriot** [reply]
@carlaLanyon Try to break down your query into small pieces. You should see whether your query is larger than 1024 characters.

```R
nchar(build_query(conversation_id = x$conversation_id))
```

I actually suggest you collecting one conversion id at a time.

**Reply by carlaLanyon** [reply]
@chainsawriot 
Hi, thanks for the answer I'm still trapped in this issue, no idea how to solve it. Any help would be great!!

I have 202 id conversations that I need to get the replay with get_all tweets. The query that made works perfectly. In order to not make it one by one I create a loop for 2150 id_conversation and rbind in order to have df with all the results. with all the results. Some id conversation doesnt have reply so I get a df empty or 0 variable. So I made a condition but still through and error. Any advice would be great!!

****Code
************** tweet_iglo_location is a data frame that contains the 202 conversation id
reply_na <- function(x){
  df <- get_all_tweets(build_query(conversation_id = tweet_na_location$conversation_id[i]),
                                      start_tweets = "2010-01-01T00:00:00Z",
                                      end_tweets = "2022-01-01T00:00:00Z",
                                      has_geo=NULL,
                                      data_path = "natamo/",
                                      bind_tweets=TRUE)
  return(df)
}

for (i in 1:2150){
  if (nrow(reply_na(i)) == 0)
    {print("Empty")} 
  else
  {
    reply <- reply_na(i) 
    reply <- reply %>% select(author_id, conversation_id, geo)
    df.reply_na<-rbind(df.reply_na,reply)
  }
}
****Error
<error/vctrs_error_subscript_oob>
Error in `stop_subscript()`:
! Can't subset columns that don't exist.
x Column `geo` doesn't exist.
******
col geo does exist

Thanks!!!
